### PR TITLE
feat: source list with pagination

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- It is highly likely that we will paginate some data in the future, hence, created pagination utilities.
- Introduced the accurate types based on the DR for integration.

https://github.com/dailydotdev/apps/assets/13744167/027f675a-b45d-4997-bee9-f582acfabf72

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/shared/src/components/search/SearchSourceItem.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export function SearchSourceItem({
+  item,
+}: SearchSourceItemProps): ReactElement {
+  if (!item) return null;
+
+  const { title, snippet, url } = item;
+
+  return (
+    <div className="w-60 laptop:w-full">
+      <ClickableText
```
Comment (rebelchris) [inline packages/shared/src/components/search/SearchSourceItem.tsx:18]: Can't remember, but was there not something wrong with this one?

**Review by rebelchris — APPROVED** [review]
Nice, looks great

File: packages/shared/src/components/search/SearchSourceItem.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/search/SearchSourceItem.tsx:18]: When wrapping it with other texts. Thankfully we don't have it here, though maybe we should now open a ticket to create the ideal link component.

File: packages/shared/src/components/search/SearchContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  onSubmit(event: React.MouseEvent, value: string): void;
+}
+
+export function SearchContainer({
+  children,
+  onSubmit,
+}: SearchContainerProps): React.ReactElement {
+  return (
+    <main className="m-auto w-full">
+      <div className="grid grid-cols-1 laptop:grid-cols-3 gap-y-6 pt-8 m-auto max-w-screen-laptop">
```
Comment (sshanzel) [inline packages/shared/src/components/search/SearchContainer.tsx:16]: @rebelchris after checking, the grid seems to be necessary for the result's UI.

Comment (rebelchris) [reply packages/shared/src/components/search/SearchContainer.tsx:16]: Ah right!! Yes remember now. But its static cols so should be fine then 🤘

**Pull request merged**